code/defines/obj/weapon.dm

```diff
@@ -81,11 +81,78 @@
 	icon_state = "cane"
 	item_state = "stick"
 	flags = CONDUCT
-	force = 10
+	force = 0
 	throwforce = 7.0
 	w_class = 4
 	matter = list(DEFAULT_WALL_MATERIAL = 50)
-	attack_verb = list("bludgeoned", "whacked", "disciplined", "thrashed")
+	//attack_verb =
+
+/obj/item/weapon/cane/afterattack(atom/A as mob|obj, mob/user as mob, proximity)
+	if (user.a_intent == I_HELP && !istype(A, /mob) && proximity)
+		A.attack_hand(user)
+
+/obj/item/weapon/cane/attack(mob/target as mob, mob/living/user as mob, var/target_zone)
+
+	var/mob/living/carbon/human/M = user

```

Comment:
This is not always true! Target specifically can be non-human, so the code should account for that. My suggestion is to run the old style attack code if the mob is non-human (a simpler mob):
```DM
/obj/item/weapon/cane/attack(mob/living/carbon/human/target, mob/living/carbon/human/user, target_zone)
	if (!istype(target) || !istype(user))
		return ..()

	// your code follows here.
```